lib/tools/system-calls.js

```diff
@@ -344,6 +345,23 @@ systemCallMethods.killAllEmulators = async function () {
   }
 };
 
+systemCallMethods.killEmulator = async function (avdName) {
+  let cmd;
+  if (system.isWindows()) {
+    cmd = `for /f "tokens=2 delims=," %a in ('tasklist /FI "imagename eq emulator*" /V /FO:csv /NH ^| findstr /r /c:".*:${avdName}.*"') do taskkill /pid %a /F`;
+  } else {
+    cmd = `pkill -9 -f 'avd ${avdName}|@${avdName}'`;
+  }
+  try {
+    log.debug(`killing avd '${avdName}'`);
+    await (B.promisify(cp.exec))(cmd);
+    await sleep(1000);
```

Comment:
In Windows, the kill command finishes before it is really finished, if you try to launch emulator exactly after comman it will fail. That is why I added this sleep.
On Jul 16, 2016 02:55, Jonathan Lipps <notifications@github.com> wrote:In lib/tools/system-calls.js:
> @@ -344,6 +345,23 @@ systemCallMethods.killAllEmulators = async function () {
>    }
>  };
>  
> +systemCallMethods.killEmulator = async function (avdName) {
> +  let cmd;
> +  if (system.isWindows()) {
> +    cmd = `for /f "tokens=2 delims=," %a in ('tasklist /FI "imagename eq emulator*" /V /FO:csv /NH ^| findstr /r /c:".*:${avdName}.*"') do taskkill /pid %a /F`;
> +  } else {
> +    cmd = `pkill -9 -f 'avd ${avdName}|@${avdName}'`;
> +  }
> +  try {
> +    log.debug(`killing avd '${avdName}'`);
> +    await (B.promisify(cp.exec))(cmd);
> +    await sleep(1000);

what's the reason for the sleep?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.